### PR TITLE
added floating point limit support

### PR DESCRIPTION
**Pull request by anopheles**
here's my pull request, note that this is a breaking change, since the limiter now assumes a float64 instead of a int64 (a version bump might be inplace :))

### CODE REVIEW

**Comment by anopheles** [comment]
there still seems to be an error with the int(max) call in 
```
return limiter.New(tbOptions).SetMax(max).SetBurst(int(max))
```
since int(0.5) is 0, every request exceeds the rate limit.

**Comment by anopheles** [comment]
@didip 
unfortunately, my change breaks compatibility. do you have any suggestions?

File: tollbooth.go [hunk 1]
```diff
@@ -3,25 +3,26 @@ package tollbooth
...
 	w.Header().Add("X-Rate-Limit-Duration", "1")
 	w.Header().Add("X-Rate-Limit-Request-Forwarded-For", r.Header.Get("X-Forwarded-For"))
 	w.Header().Add("X-Rate-Limit-Request-Remote-Addr", r.RemoteAddr)
 }
 
 // NewLimiter is a convenience function to limiter.New.
-func NewLimiter(max int64, tbOptions *limiter.ExpirableOptions) *limiter.Limiter {
-	return limiter.New(tbOptions).SetMax(max).SetBurst(int(max))
+func NewLimiter(max float64, tbOptions *limiter.ExpirableOptions) *limiter.Limiter {
+	return limiter.New(tbOptions).SetMax(max).SetBurst(int(math.Max(1, max)))
```
Comment (anopheles) [inline tollbooth.go]: this had to be done to allow rates below 1, otherwise SetBurst(0) would always result in exceeded rate limits.

**Comment by didip** [comment]
Thanks for the PR, I'll spend some time in the weekend to push a proper release.

**Pull request merged**